packages/runtime/src/templating/view.ts

```diff
@@ -95,3 +99,160 @@ export class ViewFactory<T extends INode = INode> implements IViewFactory<T> {
+  'detaching',
+  'caching',
+  'detached',
+  'unbinding',
+  'unbound'
+] as (keyof Omit<ComposableObject, 'created'>)[];
+
+export class ViewLocator implements IViewLocator {
+  private modelInstanceToBoundComponent: WeakMap<object, Record<string, Constructable>> = new WeakMap();
+  private modelTypeToUnboundComponent: Map<object, Record<string, Constructable>> = new Map();
+
+  public getViewComponentForObject(object: ComposableObject | null | undefined, viewNameOrSelector?: string | ViewSelector) {
+    if (object) {
+      const availableViews = hasAssociatedViews(object.constructor)
+        ? object.constructor.$views
+        : [];
+      const resolvedViewName = typeof viewNameOrSelector === 'function'
+        ? viewNameOrSelector(object, availableViews)
+        : this.getViewName(availableViews, viewNameOrSelector);
+
+      return this.getOrCreateBoundComponent(
+        object,
+        availableViews,
+        resolvedViewName
+      );
+    }
+
+    return null;
+  }
+
+  private getOrCreateBoundComponent(object: ComposableObject, availableViews: ITemplateDefinition[], resolvedViewName: string) {
+    let lookup = this.modelInstanceToBoundComponent.get(object);
+    let BoundComponent: Constructable | undefined;
+
+    if (!lookup) {
+      lookup = {};
+      this.modelInstanceToBoundComponent.set(object, lookup);
+    } else {
+      BoundComponent = lookup[resolvedViewName];
+    }
+
+    if (!BoundComponent) {
+      const UnboundComponent = this.getOrCreateUnboundComponent(
+        object,
+        availableViews,
+        resolvedViewName
+      );
+
+      BoundComponent = class extends UnboundComponent {
+        constructor() {
+          super(object);
+        }
+      };
+
+      lookup[resolvedViewName] = BoundComponent;
+    }
+
+    return BoundComponent;
+  }
+
+  private getOrCreateUnboundComponent(object: ComposableObject, availableViews: ITemplateDefinition[], resolvedViewName: string) {
+    let lookup = this.modelTypeToUnboundComponent.get(object.constructor);
+    let UnboundComponent: Constructable | undefined;
+
+    if (!lookup) {
+      lookup = {};
+      this.modelTypeToUnboundComponent.set(object.constructor, lookup);
+    } else {
+      UnboundComponent = lookup[resolvedViewName];
+    }
+
+    if (!UnboundComponent) {
+      UnboundComponent = CustomElement.define(
+        this.getView(availableViews, resolvedViewName),
+        class {
+          protected $scope!: IScope;
+
+          constructor(protected viewModel: ComposableObject) {}
+
+          public created(flags: LifecycleFlags) {
+            this.$scope = Scope.fromParent(flags, this.$scope, this.viewModel);
+
+            if (this.viewModel.created) {
+              this.viewModel.created(flags);
+            }
+          }
+        }
+      );
+
+      const proto = UnboundComponent.prototype;
+
+      lifecycleCallbacks.forEach(x => {
+        if (x in object) {
+          proto[x] = function() { return this.viewModel[x](); };
```

Comment:
```suggestion
          const fn = function () { return this.viewModel[x](); };
          Reflect.defineProperty(fn, 'name', { configurable: true, value: x });
          proto[x] = fn;
```

- Missing space between keyword and open paren (styling thing)
- Retain the original function name for better debugging / stack traces